src/journal/journald-native.c

```diff
@@ -198,6 +199,7 @@ void server_process_native_message(
                                         if (t) {
                                                 free(message);
                                                 message = t;
+                                                message_len = l - 8;
```

Comment:
this is not sufficient. You still are using strndup() here... which means the message string will be truncated at the first NUL byte, but you still pass the full length now, which means you basically log uninitialized memory the instant somebody sends a NUL byte...

You can simply remove the entire memory allocation here, if you pass the length anyway and instead use a pointer directly into the message...